src/lm.jl

```diff
@@ -220,7 +220,7 @@ function predict(mm::LinearModel, newx::AbstractMatrix;
         error("only :confidence and :prediction intervals are defined")
     end
     retinterval = quantile(TDist(dof_residual(mm)), (1. - level)/2) * sqrt.(retvariance)
-    hcat(retmean, retmean .+ retinterval, retmean .- retinterval)
+    (newy = retmean, lower = retmean .+ retinterval, upper = retmean .- retinterval)
```

Comment:
Just `y` would look better IMHO.

Reply:
fixed

Reply:
FWIW, R uses `fit`, `lwr` and `upr`. The two latter are clearly bad, but `fit` could also work. Opinions?

Reply:
I don't like "fit" actually, since this is a prediction from new data, not for fitted values.

Reply:
mean?

Reply:
value is not bad either

Reply:
Sorry, what kind of interval does this return when the caller provides `:prediction`?

Reply:
The prediction interval

Reply:
Can you elaborate? Does this have to do with taking or not taking uncertainty in the estimated parameters into account?

Reply:
It's the 95% confidence interval on unobserved data points, whereas the :confidence interval is the confidence on the fit. So, it incorporates the variance in the data itself

Reply:
In Bayesian stats you'd get it by simulating new values from the posterior and taking the (0.025, 0.975)  percentiles

Reply:
Thanks, rephrasing in my own words `:confidence:` gives the confidence interval for the location parameter of the conditional distribution of the response and `:prediction` gives an interval which contains the unobserved response variable with a given probability

Reply:
Yes exactly except that those words are Bayesian

Reply:
Let's go with `value` then?

Reply:
I think I prefer `prediction`.

Reply:
I also prefer prediction even if it's a little verbose.  in my own code I tend to go for `pred` but I understand that that is a convention that works for me but maybe not everyone.

Reply:
Fine with me.

Reply:
Great! The only possibly problem with that (and argument for pred or value) is that we now have `:confidence` and `:prediction` intervals, both of which return a variable named `prediction`

Reply:
sorry to be so wavering :-/

Reply:
Fine with me.

Reply:
OK it's `prediction` now and tests should pass so feel free to merge when you're happy.